static/custom_corrections.txt

```diff
@@ -0,0 +1,2 @@
+organisation->organization
```

Comment:
Wondering if the default `en-GB_to_en_US` dictionary is still picked up if we provide our custom dictionary?

https://github.com/codespell-project/codespell/blob/master/codespell_lib/data/dictionary_en-GB_to_en-US.txt

Reply:
> Wondering if the default `en-GB_to_en_US` dictionary is still picked up if we provide our custom dictionary?
> 
> https://github.com/codespell-project/codespell/blob/master/codespell_lib/data/dictionary_en-GB_to_en-US.txt

I checked it locally & looks like it doesn't. 
When I run the hook with this list, it makes some more changes.

Should I update [`static/custom_corrections.txt`](https://github.com/QuantStack/quantstack.github.io/pull/152/files/9f62721398062a9354ed84b79cf5971deaefe4a9#diff-9e7f50c19914f19118d38895b3e9672c29c23d058ab4c5fa026530b3d0661613), with respect to `en-GB_to_en_US` dictionary and push the changes or you prefer any other way?

Reply:
Since the two works listed here are already included in https://github.com/codespell-project/codespell/blob/master/codespell_lib/data/dictionary_en-GB_to_en-US.txt, maybe we can use the upstream dictionary directly?

Reply:
I tried doing that but seems like there's no direct option to use the upstream dictionary. 
I tried passing [en-GB_to_en_US dictionary url](https://github.com/codespell-project/codespell/blob/master/codespell_lib/data/dictionary_en-GB_to_en-US.txt) directly too but that doesn't work either.

For now i'd have to download & bring [en-GB_to_en_US dictionary](https://github.com/codespell-project/codespell/blob/master/codespell_lib/data/dictionary_en-GB_to_en-US.txt) in the project but I can also open a feature request upstream for built-in support for common spelling conversion dictionaries, if that sounds alright?

Reply:
Yes, i tried it with `'--builtin=dictionary_en-GB_to_en-US.txt'`, but the hook passes without making any changes. Compared to when I update the [`static/custom_corrections.txt`](https://github.com/QuantStack/quantstack.github.io/pull/152/files/9f62721398062a9354ed84b79cf5971deaefe4a9#diff-9e7f50c19914f19118d38895b3e9672c29c23d058ab4c5fa026530b3d0661613) & it makes some changes.

Reply:
Wondering if we need to pass `en-GB_to_en-US` as it seems to be its name?

https://github.com/codespell-project/codespell/blob/7f20219b05f8ed7fc48d8307c53333e8632f29a7/codespell_lib/_codespell.py#L123

Reply:
Strangely it still doesn't work with `'--builtin=en-GB_to_en-US'` 😕

Reply:
Hmm I just tried quickly and it seems to be working?

![image](https://github.com/user-attachments/assets/5f2e56a2-110b-4cca-bb15-81e5431040f7)

Reply:
> Hmm I just tried quickly and it seems to be working?

Oh I see, then I think it might be an issue working with `pre-commit`:

```
ci:
  autoupdate_schedule: monthly
  autofix_prs: true

repos:
  # Codespell hook configuration
  - repo: https://github.com/codespell-project/codespell
    rev: v2.3.0
    hooks:
      - id: codespell
        name: Spell Check with Codespell
        args:
          - '--write-changes'
          - '--ignore-words-list=cppp,normale'
          - '--dictionary=.codespell_dict'
          - '--builtin=en-GB_to_en-US'
        # Exclude package-lock.json and all .svg files
        exclude: 'package-lock\.json|\.svg$'
```

i.e. when I'm doing `pre-commit run --all-files` with the above `.pre-commit-config.yaml`.

Reply:
It seems to be working after removing this line:

```
          - '--dictionary=.codespell_dict'
```

![image](https://github.com/user-attachments/assets/9f9e4ff3-10d0-41e2-b2b6-7f786c201345)

Reply:
Oh Oops! 👀
Yes it does. So sorry to bother you with this, this was my bad.